cbits/tcdrain.c

```diff
@@ -0,0 +1,21 @@
+/* Provide tcdrain() if it is missing (as on some Android versions). */
+
+#include "HsUnixConfig.h"
+#ifndef HAVE_TCDRAIN
+
+#include "ghcplatform.h"
+
+#ifdef linux_android_HOST_OS
+#include <sys/ioctl.h>
+
+int tcdrain(int fd)
```

Comment:
I don't like this, as we'd be defining POSIX defined `tcdrain`/`ctermid` identifiers into a global namespace for which we have no ownership. We shouldn't need any cbits at all for providing a graceful fallback implementation.